metadata-io/src/main/java/com/linkedin/metadata/system_info/collectors/SpringComponentsCollector.java

```diff
@@ -0,0 +1,198 @@
+package com.linkedin.metadata.system_info.collectors;
+
+import static com.linkedin.metadata.system_info.SystemInfoConstants.*;
+
+import com.linkedin.metadata.system_info.ComponentInfo;
+import com.linkedin.metadata.system_info.ComponentStatus;
+import com.linkedin.metadata.system_info.SpringComponentsInfo;
+import com.linkedin.metadata.version.GitVersion;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.TimeUnit;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+/**
+ * Collects information about Spring components (GMS, MAE Consumer, MCE Consumer).
+ *
+ * <p>DEPLOYMENT MODES: DataHub can be deployed in two ways: 1. Embedded mode (default quickstart):
+ * All components run in single GMS container 2. Separate containers: Each component runs in its own
+ * container for better scaling
+ *
+ * <p>CURRENT LIMITATION: This collector currently returns placeholder data for MAE/MCE consumers to
+ * avoid a circular dependency issue that occurs in embedded mode:
+ *
+ * <p>The Problem: - System-info endpoint (/openapi/v1/system-info) calls this collector - Collector
+ * tries to fetch consumer info by calling same endpoint on consumer URLs - In embedded mode,
+ * consumer URLs point back to the same GMS process - This creates infinite recursion / timeout
+ * after 5 seconds
+ *
+ * <p>The Solution: - Return placeholder data instead of making HTTP calls - Maintains API structure
+ * while avoiding circular dependency - Allows system-info endpoint to work properly in embedded
+ * mode
+ *
+ * <p>FUTURE IMPROVEMENTS: A proper solution would detect deployment mode and handle accordingly: -
+ * Embedded mode: Query internal component state directly (no HTTP calls) - Separate containers:
+ * Make HTTP calls to external consumer service endpoints
+ *
+ * <p>The placeholder approach provides a stable foundation for implementing this detection logic.
+ */
+@Slf4j
+@Component
+@RequiredArgsConstructor
+public class SpringComponentsCollector {
```

Comment:
I removed fetchRemoteComponentInfo - it is not used and it was adding to non covered lines. 

it is still here https://github.com/datahub-project/datahub/blob/ba373d73079bc8e4fe60e9a2e591003eb7294e26/metadata-io/src/main/java/com/linkedin/metadata/system_info/collectors/SpringComponentsCollector.java#L210